lake/index/expr.go

```diff
@@ -112,15 +126,19 @@ func compareExpr(kv index.KeyValue, op string) (expr, error) {
 	}, nil
 }
 
-func matchFieldRule(rules []Rule, in index.KeyValue) Rule {
+func matchFieldRule(rules []Rule, in index.KeyValue) (index.KeyValue, Rule) {
 	for _, rule := range rules {
 		// XXX support indexes with multiple keys #3162
 		// and other rule types.
 		if fr, ok := rule.(*FieldRule); ok && in.Key.Equal(fr.Fields[0]) {
-			return rule
+			out := index.KeyValue{
+				Key:   append(field.New("key"), in.Key...),
+				Value: in.Value,
+			}
+			return out, rule
```

Comment:
Unless you find this clearer with `out`, of course.
```suggestion
			return index.KeyValue{
				Key:   append(field.New("key"), in.Key...),
				Value: in.Value,
			}, rule
```